cmd/cluster/create.go

```diff
@@ -195,20 +202,22 @@ func CreateCluster(ctx context.Context, opts Options) error {
 		if len(infraID) == 0 {
 			infraID = fmt.Sprintf("%s-%s", opts.Name, utilrand.String(5))
 		}
-		opt := awsinfra.CreateInfraOptions{
-			Region:             opts.Region,
-			InfraID:            infraID,
-			AWSCredentialsFile: opts.AWSCredentialsFile,
-			Name:               opts.Name,
-			BaseDomain:         opts.BaseDomain,
-			AdditionalTags:     opts.AdditionalTags,
-		}
-		infra, err = opt.CreateInfra(ctx)
-		if err != nil {
-			return fmt.Errorf("failed to create infra: %w", err)
+
+		if opts.Platform == string(hyperv1.AWSPlatform) {
+			opt := awsinfra.CreateInfraOptions{
+				Region:             opts.Region,
+				InfraID:            infraID,
+				AWSCredentialsFile: opts.AWSCredentialsFile,
+				Name:               opts.Name,
+				BaseDomain:         opts.BaseDomain,
+				AdditionalTags:     opts.AdditionalTags,
+			}
+			infra, err = opt.CreateInfra(ctx)
+			if err != nil {
+				return fmt.Errorf("failed to create infra: %w", err)
+			}
 		}
 	}
-
 	var iamInfo *awsinfra.CreateIAMOutput
 	if len(opts.IAMJSON) > 0 {
```

Comment:
wouldn't this whole thing need to within `if opts.Platform == string(hyperv1.AWSPlatform) {`

This lgtm overall. Either here or in a follow up since we are growing in platforms I think we should consider a more structure flow with may be specific functions per providers.